gdx/src/com/badlogic/gdx/maps/tiled/BaseTmxMapLoader.java

```diff
@@ -83,6 +85,16 @@ public Array<AssetDescriptor> getDependencies (String fileName, FileHandle tmxFi
 		return getDependencyAssetDescriptors(tmxFile, textureParameter);
 	}
 
+	/** Gets a map of the object ids to the {@link MapObject} instances. The returned map is a copy of the internal map. Returns
+	 * null if {@link #loadTiledMap(FileHandle, Parameters, ImageResolver)} has not been called yet. The internal map is reset when
+	 * the load tiled map method is called.
+	 *
+	 * @return the map of the object ids to the {@link MapObject} instances, or null if the load tiled map method has not been
+	 *         called yet. */
+	public @Null IntMap<MapObject> getIdToObject () {
+		return idToObject == null ? null : new IntMap<>(idToObject);
```

Comment:
Why a copy?

Reply:
It returns a copy so that the internal state of the map can't be changed from the outside. The user should call this method only once each time the loadTiledMap method is called so that it doesn't incur much runtime penalty.

Reply:
I don't have a strong opinion about this though if you think it should just return the internal map instead of a copy

Reply:
I think you make a great argument for why it shouldn't return a copy. My thought was to protect the internal state, but after reading your explanation, I agree there's not much point since the map is reallocated every time the load tiled map method is called. I've changed it.